app/models/portfolio.rb

```diff
@@ -21,6 +21,17 @@ class Portfolio < ApplicationRecord
     end
   end
 
+  before_undiscard do
+    if portfolio_items.with_discarded.discarded.map(&:undiscard).any? { |result| result == false }
```

Comment:
I'm not seeing the timestamp part discussed in the Jira ticket. https://projects.engineering.redhat.com/browse/SSP-169

Should be item 5.

Reply:
Added! added a test as well. Picked 30 seconds arbitrarily - as that seems to be a low yet not too high of a number. Open to suggestions @syncrou